woocommerce/woocommerce-android
# Variant visibility switch

Fixes #2194. This PR adds all the necessary infrastructure to make variations editable.

| Light   | Dark |
|----------|-------------|
| ![Screenshot_1596034524](https://user-images.githubusercontent.com/1522856/88816393-a32e5180-d1bc-11ea-937a-817d814e11c1.png) | ![Screenshot_1596034545](https://user-images.githubusercontent.com/1522856/88816495-c22ce380-d1bc-11ea-968d-3caf42917044.png) |

**To test**
1. Open a variable product
2. Tap on Variations property
3. Select a variation and open Variation details
4. Tap on the visibility switch to change it
5. **Notice the icon changed and an Update button appeared**
6. Tap on the back button
7. **Notice a discard dialog is shown**
8. Tap on Discard
9. Open the same variation again
10. **Notice the visibility has not changed**
11. Tap on the switch and tap on the Update button
12. **Notice a progress dialog is shown**
13. After the progress dialog disappears, **notice that the Update button is gone**
14. Tap on the back button
15. **Notice that no discard dialog appears and that the visibility is shown in the list item**
16. Open the same variation again
17. **Notice the visibility has been preserved**
